src/TokenSchedule.FluentValidation/ScheduleItemValidator.cs

```diff
@@ -0,0 +1,19 @@
+﻿using FluentValidation;
+using TokenSchedule.FluentValidation.Models;
+
+namespace TokenSchedule.FluentValidation
+{
+    public class ScheduleItemValidator : AbstractValidator<IValidatedScheduleItem>
+    {
+        public ScheduleItemValidator()
+        {
+            RuleFor(item => item)
+                .Cascade(CascadeMode.Stop)
+                .NotNull()
+                .Must(item => item.Ratio > 0)
+                .WithMessage("Ratio must be positive.")
+                .Must(item => !item.FinishDate.HasValue || item.StartDate < item.FinishDate.Value)
```

Comment:
lets do it using when?

Reply:
When we use `When()` function, we need to suppress nullable of `FinishDate` via `!`, which is not best practice.
![image](https://github.com/user-attachments/assets/4ddf93fb-e940-4271-bbb5-df95e187122c)

Reply:
I think in this case we can use `!` - the when make it safe